actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -327,13 +327,14 @@ def button_to(name = nil, options = nil, html_options = nil, &block)
         html_options = convert_options_to_data_attributes(options, html_options)
         html_options["type"] = "submit"
 
-        button = if block_given?
-          content_tag("button", html_options, &block)
+        content = if block_given?
+          capture(&block)
         else
-          html_options["value"] = name || url
-          tag("input", html_options)
+          name || url
         end
```

Comment:
Feels like content assignment would be clearer as a ternary and then defining button right after:

```ruby
content = block_given? ? capture(&block) : name || url
button  = content_tag("button", content, html_options, &block)
```